src/DeckEditor.jsx

```diff
@@ -0,0 +1,472 @@
+import { useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router';
+import useDeckStore from './stores/deck';
+import { useShallow } from 'zustand/react/shallow';
+
+export default function DeckEditor() {
+  const navigate = useNavigate();
+  const { deckId = 'standard' } = useParams();
+
+  const { decks: decksById, ownedShopCards } = useDeckStore(
+    useShallow(s => ({ decks: s.decks, ownedShopCards: s.ownedShopCards }))
+  );
+
+  const { addToLoadoutFor, removeFromLoadoutFor } = useDeckStore(
+    useShallow(s => ({
+      addToLoadoutFor: s.addToLoadoutFor,
+      removeFromLoadoutFor: s.removeFromLoadoutFor,
+    }))
+  );
+
+  const { setActiveDeck, buildAndShuffle } = useDeckStore(
+    useShallow(s => ({
+      setActiveDeck: s.setActiveDeck,
+      buildAndShuffle: s.buildAndShuffle,
+    }))
+  );
+
+  const { getDisplayName, getDisplayDesc } = useDeckStore(
+    useShallow(s => ({
+      getDisplayName: s.getDisplayName,
+      getDisplayDesc: s.getDisplayDesc,
+    }))
+  );
+
+  const { openEditor, setEditorField, saveEditor, cancelEditor, editor } = useDeckStore(
+    useShallow(s => ({
+      openEditor: s.openEditor,
+      setEditorField: s.setEditorField,
+      saveEditor: s.saveEditor,
+      cancelEditor: s.cancelEditor,
+      editor: s.editor,
+    }))
+  );
+
+  const currentDeck = decksById[deckId] || { id: deckId, loadout: [] };
+  const loadoutIds = new Set(currentDeck.loadout.map(card => card.id));
+  const loadoutCount = currentDeck.loadout.length;
+
+  const name = getDisplayName(deckId);
+  const desc = getDisplayDesc(deckId);
+  const isEditing = editor.isOpen && editor.deckId === deckId;
+
+  const [filterQuery, setFilterQuery] = useState('');
+  const filteredOwned = useMemo(() => {
+    const term = filterQuery.trim().toLowerCase();
+    if (!term) return ownedShopCards;
+    return ownedShopCards.filter(
+      c =>
+        (c.name || '').toLowerCase().includes(term) ||
+        (c.desc || '').toLowerCase().includes(term)
+    );
+  }, [filterQuery, ownedShopCards]);
+
+  const filteredOwnedNotInDeck = useMemo(
+    () => filteredOwned.filter(c => !loadoutIds.has(c.id)),
+    [filteredOwned, loadoutIds]
+  );
+
+  const handleAdd = id => addToLoadoutFor(deckId, id);
+  const handleRemove = id => removeFromLoadoutFor(deckId, id);
+
+  const handleSaveDeck = () => {
+    saveEditor();
+    buildAndShuffle(deckId);
+    navigate('/deck-select');
+  };
+
+  return (
+    <div style={pageWrapStyle}>
```

Comment:
Players shouldn't be able to edit the deck when they select it. They also shouldn't be able to perform any bulk edits. I really appreciate all the work you've added here. But really all we need right now is the ability to add standard cards to the card shop and the ability to add those cards into the cards array within the deck store. That is enough for us to modify the deck as the player progresses their run. Maybe in the future we will add a bulk edit window like this. But for now the only place we need to add a new card to the deck from is the shop and the only way you can see the cards is by looking through your entire deck in the deck viewer page.